src/components/Nodes/ClientConfig/VersionListItem.jsx

```diff
@@ -58,12 +67,13 @@ export default class VersionListItem extends Component {
       let localRelease
       try {
         localRelease = await client.download(release, downloadProgress => {
-          this.setState({ downloadProgress })
+          if (this._isMounted) this.setState({ downloadProgress })
         })
       } catch (error) {
         handleDownloadError(error)
       }
-      this.setState({ isDownloading: false, downloadProgress: 0 })
+      if (this._isMounted)
+        this.setState({ isDownloading: false, downloadProgress: 0 })
```

Comment:
linter should bark about multi-line conditionals without curly brackets, though. will check into that 👀